### DISCUSSION TITLE
Fresh clone fails one test with uninitialized constant Puma::Configuration

### DISCUSSION THREAD

**Discussion by CaioCSdev** [puma/puma#3231] [Q&A]
Hello all :wave: 

I want to help Puma and, if I can, make some contributions.

To start, I've cloned the repo and followed the instructions on the CONTRIBUTING.md. Everything going fine until I try to run the tests, I get a strange error on `test/test_integration_cluster.rb` `TestIntegrationCluster#test_worker_timeout` (I have already increased the file limits to 4000)

```
TestIntegrationCluster#test_worker_timeout:
NameError: uninitialized constant Puma::Configuration
    /puma/test/test_integration_cluster.rb:207:in `test_worker_timeout'
    /puma/test/helper.rb:91:in `block (4 levels) in run'
    /usr/local/lib/ruby/3.2.0/timeout.rb:189:in `block in timeout'
    /usr/local/lib/ruby/3.2.0/timeout.rb:196:in `timeout'
    /puma/test/helper.rb:89:in `block (3 levels) in run'
```

**How come uninitialized constant Puma::Configuration ?**

I manage to bypass the error by adding `require "puma/configuration"` but it shouldn't be needed right?

To check if the issue is not with my machine I made a dockerfile that only runs the failing test

```Dockerfile
FROM ruby

WORKDIR /puma

RUN git clone https://github.com/puma/puma.git .

RUN apt update && apt install ragel

RUN bundle install

RUN bundle exec rake compile

CMD ["bundle", "exec", "m", "test/test_integration_cluster.rb:207"]
```

just build and run the Dockerfile, the error should show up

Thanks for the awesome project, I hope I can contribute :heart:

**Comment by MSP-Greg** [comment]
@CaioCSdev

Thanks.

Re the error you're seeing, I've got a branch with that fix, but I haven't pushed it yet.

When one runs the **whole test suite**, Minitest loads all the test files at once.  So, there is a test file that requires `puma/configuration`.

But, if one runs a single test file, there can be missing require statements.  Many of files in the test suite have been around for several years, and the test code using `Puma::Configuration` was added about a year ago...

**Comment by MSP-Greg** [comment]
See PR #3233, commit ['test_integration_cluster.rb - add require for single run'](https://github.com/puma/puma/pull/3233/commits/768a166fcc39ad4398114cd306e376774278149a)
